go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -110,6 +110,10 @@ func (oc *Controller) addEgressFirewall(egressFirewall *egressfirewallapi.Egress
 		return errList
 	}
 
+	if nsInfo.addressSet == nil {
+		return []error{fmt.Errorf("unable to add egress firewall policy, namespace: %s has no address set", egressFirewall.Namespace)}
```

Comment:
So in this case, if this fails, it's not going to get retried.

I guess maybe that's not a new problem since this function might fail in other places anyway...

Reply:
That's a good point. However, what is our answer to when these things happen (especially at scale) short of writing a proper reconcile loop for each object type?

Reply:
I mean, we need proper reconciliation everywhere. But that's a broader, pre-existing problem.

Reply:
for the egressFirewall case it is a little easier, since we don't add anything to the address_set we could just generate the name using `getIPv4ASName` and `getIPv6ASName` and add it to the match case in the logical_router policy.

Reply:
well for this code we only need the name, but in the end thats going to match some address set in OVN right with and ACL. if that address set doesnt exist thats a problem. Maybe I should try to create the address set here as well if it is missing?

Reply:
Actually @JacobTanenbaum just reminded me of the discussion we had last week on breaking apart dependencies on address set for various resources, egress firewall being one of them. The thing is that OVN doesn't really need the address set to configure the ACL in NB. ovn-controller will just hold off on programming any flows for that ACL until the address set is created. So all we really need are the names here of the address sets. I think the right way forward is to not get any lock and just generate the names.